src/XrdCms/XrdCmsRedirLocal.cc

```diff
@@ -113,6 +118,31 @@ int XrdCmsRedirLocal::Locate(XrdOucErrInfo &Resp, const char *path, int flags,
                         XrdOucEnv *EnvInfo) {
   int rcode = 0;
   if (nativeCmsFinder) {
+    // check if path contains localroot to know if potential redirection loop
+    // is happening. If yes, remove localroot from path, then rerun default
+    // locate with regular path and return to client
+    // localroot must be larger than 1 to avoid always being triggered by "/"
+    if (localroot.size() > 1 && strncmp(path, localroot.c_str(), localroot.size()) == 0)
+    {
+      // now check if localhost was tried before, to make sure we're handling
+      // the redirection loop
+      int param = 0; // need it to get Env
+      std::string envInfo(EnvInfo->Env(param)); // get already tried hosts
+      std::string searchPattern = "&tried=localhost"; //search for this pattern
```

Comment:
Thanks, corrected. But we're now searching for "tried=localhost" in the full envInfo. Otherwise, it might be missed, if other params are passed first.